util/topgen/templates/toplevel_racl_pkg.sv.tpl

```diff
@@ -0,0 +1,82 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+${gencmd}
+
+package top_racl_pkg;
+  // Number of RACL policies used
+  parameter int unsigned NrRaclPolicies = ${racl_config['nr_policies']};
+
+  // Number of RACL bits transferred
+  parameter int unsigned NrRaclBits = 4;
+
+  // Number of CTN UID bits transferred
+  parameter int unsigned NrCtnUidBits = 8;
+
+  // RACL role type binary encoded
+  typedef logic [NrRaclBits-1:0] racl_role_t;
+
+  // CTN UID assigned the bus originator
+  typedef logic [NrCtnUidBits-1:0] ctn_uid_t;
+
+  // RACL permission and one-hot encoded role vector
+  typedef logic [(2**NrRaclBits)-1:0] racl_role_vec_t;
+
+  // RACL policy vector containing a read and write permission
+  typedef struct packed {
+    racl_role_vec_t read_perm;
+    racl_role_vec_t write_perm;
+  } racl_policy_t;
+
+  // RACL policy vector for distributing RACL policies from the RACL widget to the subscribing IP
+  typedef racl_policy_t [NrRaclPolicies-1:0] racl_policy_vec_t;
+
+  // RACL information logged in case of a denial
+  typedef struct packed {
+    racl_role_t racl_role;
+    ctn_uid_t   ctn_uid;
+    // 0: Write access, 1: Read access
+    logic       write_read;
```

Comment:
Not a big deal if this name appears in lots of other places. But something like `read_not_write` might be more obvious at a glance?